src/main/java/org/mitre/bonnie/cqlTranslationServer/TranslationResource.java

```diff
@@ -148,6 +153,23 @@ public Response cqlPackageToElmPackage(
       libraryManager.getLibrarySourceLoader().clearProviders();
     }
   }
+  
+  private void registerModelInfos(FormDataMultiPart pkg) {
+    CustomModelInfoProvider mip = null;
+    for( BodyPart part : pkg.getBodyParts() ) {
```

Comment:
Question - seems like we would allow multiple custom modelinfos to be provided with a request here? Is that something we want to allow and if so, what is downstream translator behavior if there are more than 1 modelinfo in effect? Just wanted to confirm that's a scenario supported by the translator.

Reply:
You are correct that the CQL translator looks up the appropriate modelinfo based on the using statement in the CQL. It is cross-referenced against the name + version in the root element of the modelinfo XML. You are also correct that this translation API now supports passing more than one custom model in a request. I don't anticipate consumers of the service will use more than one custom model at a time, but I also don't see any particular reason to block them from doing so. The only place I forsee multiple models getting a user into trouble is if they tried to load the same model name + version more than once in the same request. In that case, the translator would throw an Exception. I didn't write a test case for that, but I could if folks are interested.